src/core/utils.ts

```diff
@@ -187,6 +191,88 @@ export function hasVisibleLegendItems(options: Highcharts.Options) {
   });
 }
 
+export function shouldShowSecondaryLegend(options: Highcharts.Options) {
+  // Note: While native Highcharts supports multiple legends, this
+  // implementation simplifies to at most 2 legends:
+  // - One group for all axes with opposite=false (primary axes)
+  // - One group for all axes with opposite=true (secondary axes)
+  //
+  // Important: A secondary legend should only be shown when BOTH primary and secondary
+  // series exist. This prevents abuse of the secondary legend API to force right-hand
+  // alignment when there's no actual dual-legend layout.
+  //
+  // Why we target yAxis only:
+  // In typical chart usage, dual-axis layouts use the y-axis for different scales
+  // (e.g., temperature on left, humidity on right). The x-axis is shared across all series.
+  // While Highcharts technically supports opposite x-axes, this is rare in practice and
```

Comment:
Oh, that is actually interesting. I think what you are referring to is called `chart.inverted`. Can we do some `isInverted ? xAxis : yAxis` instead? It feels that supporting the inverted orientation should not be hard.

The opposite is also interesting, though. It allow to render the axis to the opposite side of the chart, which should mean that the primary and secondary legends should exchange places. I think it is fine to not support it with the initial implementation, but let's document in comments properly.